### PR TITLE
[PVE-011] Implement OZ ReentrancyGuard

### PR DESCRIPTION
**Pull request by haythem96**
This PR apply the necessary reentrancy prevention for the listed below contract's functions by  implementing the OZ `nonReentrant` modifier.

- MinterBridge::bridgeTransferFrom()
- MarketContractProxy::batchRedeem()
- MarketContractProxy::mintPositionTokens() 
- MarketCollateralPool::mintPositionTokens()

### CODE REVIEW

File: contracts/honeylemon/MarketContractProxy.sol [hunk 1]
```diff
@@ -545,13 +546,13 @@ contract MarketContractProxy is Ownable {
         shortToken.transfer(getUserAddressOrDSProxy(shortTokenRecipient), qtyToMint);
 
         emit PositionTokensMinted(
+            qtyToMint,
             addressToMarketId[address(latestMarketContract)], // MarketID
```
Comment (haythem96) [inline contracts/honeylemon/MarketContractProxy.sol:550]: Modified the order here in the event to fix the `CompilerError: Stack too deep, try removing local variables.` error.

File: contracts/honeylemon/MarketContractProxy.sol [hunk 2]
```diff
@@ -92,13 +93,13 @@ contract MarketContractProxy is Ownable {
     //// EVENTS ////
     ////////////////
     event PositionTokensMinted(
+        uint256 qtyToMint,
         uint indexed marketId,
```
Comment (haythem96) [inline contracts/honeylemon/MarketContractProxy.sol:97]: Same here.

File: subgraph/subgraph.yaml [hunk 3]
```diff
@@ -45,7 +45,7 @@ dataSources:
               - name: MarketContractProxy
                 file: ../build/contracts/MarketContractProxy.json
           eventHandlers:
-              - event: PositionTokensMinted(indexed uint256,string,indexed address,address,indexed address,address,uint256,address,address,address,uint256)
+              - event: PositionTokensMinted(uint256,indexed uint256,string,indexed address,address,indexed address,address,address,address,address,uint256)
                 handler: handlePositionTokensMinted
```
Comment (haythem96) [inline subgraph/subgraph.yaml:49]: Needed to update the graph as the event was modified.

File: contracts/honeylemon/MarketContractProxy.sol (hunk 1 above)
Comment (chrismaree) [reply contracts/honeylemon/MarketContractProxy.sol:550]: this change will break the graph implementation @andrei-anisimov. We will need to update that accordingly when this is merged in.

File: contracts/honeylemon/MinterBridge.sol [hunk 4]
```diff
@@ -24,6 +25,8 @@ contract MinterBridge is Ownable {
     address public MARKET_CONTRACT_PROXY_ADDRESS;
     address public ERC20_BRIDGE_PROXY_ADDRESS;
 
+    constructor() public ReentrancyGuard() {}
```
Comment (chrismaree) [inline contracts/honeylemon/MinterBridge.sol:28]: does the constructor need this?

File: subgraph/subgraph.yaml (hunk 3 above)
Comment (chrismaree) [reply subgraph/subgraph.yaml:49]: @andrei-anisimov this is the only update needed, right?

File: contracts/honeylemon/MinterBridge.sol (hunk 4 above)
Comment (haythem96) [reply contracts/honeylemon/MinterBridge.sol:28]: mm yeah I think so!

**Pull request merged**